api/nodes/permissions.py

```diff
@@ -112,6 +129,21 @@ def has_object_permission(self, request, view, obj):
         else:
             return obj.has_permission(auth.user, osf_permissions.ADMIN)
 
+class AdminContributorOrPublic(permissions.BasePermission):
+
+    acceptable_models = (AbstractNode, DraftRegistration)
+
+    def has_object_permission(self, request, view, obj):
+        """
+        To make changes, user must be an admin contributor. Admin group membership is not sufficient.
+        """
+        assert_resource_type(obj, self.acceptable_models)
+        auth = get_user_auth(request)
+        if request.method in permissions.SAFE_METHODS:
+            return obj.is_public or obj.can_view(auth)
```

Comment:
It's not entirely redundant, as you allude to in your other comment, because looking through Node.can_view(...), if I'm viewing the node through an anonymous, private link but that link isn't attached to this node, then this would return False.  But we want the is_public check to bypass everything.  The Node.can_view() method may change in the future, but if it's public, regardless of any other conditionals in can_view, we want them to be able to view this.

Regarding the use of ternary statements, my personal preference is to avoid them. It's less lines, but sometimes readability is sacrificed, and I like being able to quickly put a breakpoint in there if need be.   If it's a very succinct ternary, I may use it.